### PR TITLE
Reworked Categorization AI Training Logs

### PR DESCRIPTION
**Pull request by MohamedAmineDHIAB**
- [x] Removed: logs showing wrong results for test accuracy

```diff 
- logger.info(f'test_loss: {np.mean(train_losses):.3f}, test_acc: {np.nanmean(train_loss):.3f}')
```

- [x] Added: computation and training logs for `f1_score` and `accuracy_score`

```diff 
+            f1_score = f1_metric.compute(predictions=predictions_list, references=ground_truth_list, average="macro")[
+                "f1"
+            ]  # compute f1 score
+            acc_score = acc_metric.compute(predictions=predictions_list, references=ground_truth_list)[
+                "accuracy"
+            ]  # compute accuracy score
```

```diff 
+ logger.info(
+                f'Epoch: {epoch + 1} | '
+                f'Train Loss: {np.mean(train_loss):.3f} | '
+                f'Train F1: {f1_score:.3f} | '
+                f'Train Accuracy: {acc_score:.3f} |'
+          )
```

- [x] Reworked: some training logs (logging number of `Training Documents` and `Training Pages` instead of just logging `Training Examples` which is unintuitive)

### CODE REVIEW

File: konfuzio_sdk/trainer/document_categorization.py [hunk 1]
```diff
@@ -1286,14 +1293,28 @@ def _fit_classifier(
...
+        acc_metric = evaluate.load("accuracy")
+        logger.info('Begin fitting')
         best_valid_loss = float('inf')
         train_loss = float('inf')
         for epoch in range(n_epochs):
-            train_loss = self._train(train_examples, loss_fn, optimizer)  # train epoch
+            train_loss, predictions_list, ground_truth_list = self._train(
+                train_examples, loss_fn, optimizer
+            )  # train epoch
+            f1_score = f1_metric.compute(predictions=predictions_list, references=ground_truth_list, average="macro")[
```
Comment (iftwigs) [inline konfuzio_sdk/trainer/document_categorization.py:1305]: is it possible to use `konfuzio_sdk.evaluate.CategorizationEvaluation` here and in accuracy calculation instead?

File: konfuzio_sdk/trainer/document_categorization.py [hunk 2]
```diff
@@ -1311,12 +1332,10 @@ def _fit_classifier(
             if patience_counter > patience:
                 logger.info('lost patience!')
                 break
-        logger.info('training finished, begin testing')
+        logger.info('Training finished. Loading best model')
```
Comment (iftwigs) [inline konfuzio_sdk/trainer/document_categorization.py:1335]: Loading the best model (sorry for nitpicking)

**Review by zypriafl — APPROVED** [review]

File: konfuzio_sdk/trainer/document_categorization.py (hunk 1 above)
Comment (MohamedAmineDHIAB) [reply konfuzio_sdk/trainer/document_categorization.py:1305]: Yes, I was thinking of that in the beginning but since `train_examples` and `predictions` are on the pages level and not the documents I decided to directly compute the f1 score with `evalute` (while `CategorizationEvaluation` requires the documents as a SDK Document class)
